stock/src/main/java/nl/tudelft/wdm/group1/stock/StockItem.java

```diff
@@ -1,22 +1,48 @@
 package nl.tudelft.wdm.group1.stock;
 
+import jdk.nashorn.internal.objects.annotations.Constructor;
+
 import java.util.UUID;
 
 public class StockItem {
     private UUID id;
+    private int stock;
 
     public StockItem() {
-        id = UUID.randomUUID(); // TODO: Remove this line as soon as the non-default constructor is created
+
+    }
+
+    public StockItem(final int stock) {
+        this.id = UUID.randomUUID();
+        this.stock = stock;
     }
 
     public UUID getId() {
         return id;
     }
 
+    public int getStock() { return this.stock; }
+
+    public void addStock(final int amount) {
+        if (amount > 0) this.stock += amount; // TODO: maybe create a new exception for this?
+    }
+
+    public void subtractStock(final int amount) throws InsufficientStockException{
+        if (amount > this.stock) {
+            throw new InsufficientStockException("Insufficient stock");
+        }
+
+        if (amount > 0) {
+            this.stock -= amount;
+        }
```

Comment:
else?

Reply:
It does nothing if you subtract a negative number. I omitted the else

Reply:
See previous comment about throwing an exception